source/module_base/formatter.h

```diff
@@ -1,43 +1,307 @@
+/**
+ * @file formatter.h
```

Comment:
should we need a page on abacus.deepmodeling.com to show users how to use formatter?

Reply:
after merging this PR, I will prepare such a doc.